pilot/pkg/networking/core/route/route.go

```diff
@@ -1242,6 +1242,14 @@ func BuildDefaultHTTPInboundRoute(clusterName string, operation string) *route.R
 		// gRPC requests time out like any other requests using timeout or its default.
 		GrpcTimeoutHeaderMax: Notimeout,
 	}
+	if proxy.VersionGreaterOrEqual(&model.IstioVersion{Major: 1, Minor: 24, Patch: -1}) && features.EnableInboundRetryPolicy {
```

Comment:
should we provide a function, i believe @keithmattix 's pr should add this check too

Reply:
I can add a function

Reply:
@hzxuzhonghu I'm a bit confused about why validation needs a version check

Reply:
Not validation, but the place where it build route per proxy